lib/active_publisher/async/in_memory_adapter/consumer_thread.rb

```diff
@@ -58,30 +58,27 @@ def start_thread
               # If the queue is empty, we should continue to update to "last_tick_at" time.
               next if current_messages.nil?
 
+              # We only look at active publisher messages. Everything else is dropped.
+              current_messages.select! { |message| message.is_a?(::ActivePublisher::Message) }
```

Comment:
what does this do? (just wondering why it is necessary)

Reply:
This is a sanity check to drop any non active publisher messages. I'm being defensive here since a `nil` or otherwise bad message could trigger this code path to spin forever.